### PR TITLE
Added support for format parameter on Share Get Permission

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2024-11-04/file.json [hunk 1]
```diff
@@ -6543,6 +6546,17 @@
...
+        },
+        "format": {
+          "type": "string",
+          "enum": [
+            "SDDL",
+            "binary"
+          ],
+          "x-ms-enum": {
+            "name": "FilePermissionKeyFormat",
+            "modelAsString": false
```
Comment (JeffreyRichter) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2024-11-04/file.json:6558]: If you use modelAsString:false, then the set of enum values can NEVER change. Is this what you want?

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2024-11-04/file.json [hunk 2]
```diff
@@ -6981,6 +6995,23 @@
...
+      "x-ms-parameter-location": "method",
+      "required": false,
+      "type": "string",
+      "enum": [
+        "SDDL",
+        "binary"
+      ],
+      "x-ms-enum": {
+        "name": "FilePermissionKeyFormat",
+        "modelAsString": false
```
Comment (JeffreyRichter) [inline specification/storage/data-plane/Microsoft.FileStorage/stable/2024-11-04/file.json:7012]: Same comment as above

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2024-11-04/file.json (hunk 1 above)
Comment (seanmcc-msft) [reply specification/storage/data-plane/Microsoft.FileStorage/stable/2024-11-04/file.json:6558]: What can't we add additional enum values in the future?

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2024-11-04/file.json (hunk 2 above)
Comment (seanmcc-msft) [reply specification/storage/data-plane/Microsoft.FileStorage/stable/2024-11-04/file.json:7012]: Same.

File: specification/storage/data-plane/Microsoft.FileStorage/stable/2024-11-04/file.json (hunk 1 above)
Comment (JeffreyRichter) [reply specification/storage/data-plane/Microsoft.FileStorage/stable/2024-11-04/file.json:6558]: No, if you want the flexibility to add more in the future, then use modelAsString:true (not false).

**Review by jalauzon-msft — APPROVED** [review]

**Pull request merged**